telegram/ext/_commandhandler.py

```diff
@@ -223,71 +261,56 @@ class PrefixHandler(CommandHandler):
 
     """
 
-    # 'prefix' is a class property, & 'command' is included in the superclass, so they're left out.
-    __slots__ = ('_prefix', '_command', '_commands')
+    __slots__ = ('_prefixes', '_combinations')
 
     def __init__(
         self,
-        prefix: SLT[str],
-        command: SLT[str],
+        prefix: SLTS[str],
+        command: SLTS[str],
         callback: Callable[[Update, CCT], RT],
         filters: filters_module.BaseFilter = None,
         run_async: Union[bool, DefaultValue] = DEFAULT_FALSE,
     ):
 
-        self._prefix: List[str] = []
-        self._command: List[str] = []
-        self._commands: List[str] = []
+        self._prefixes: Set[str] = set()
+        self._combinations: Set[str] = set()
 
         super().__init__(
-            'nocommand',
+            command,
             callback,
             filters=filters,
             run_async=run_async,
         )
 
-        self.prefix = prefix  # type: ignore[assignment]
-        self.command = command  # type: ignore[assignment]
-        self._build_commands()
+        self.prefixes = prefix  # type: ignore[assignment]
 
     @property
-    def prefix(self) -> List[str]:
+    def prefixes(self) -> Set[str]:
         """
-        The prefixes that will precede :attr:`command`.
+        The prefixes that will precede :attr:`commands`.
 
         Returns:
-            List[:obj:`str`]
+            Set[:obj:`str`]
         """
-        return self._prefix
+        return self._prefixes
 
-    @prefix.setter
-    def prefix(self, prefix: Union[str, List[str]]) -> None:
+    @prefixes.setter
+    def prefixes(self, prefix: SLTS) -> None:
         if isinstance(prefix, str):
-            self._prefix = [prefix.lower()]
+            self._prefixes = {prefix.lower()}
         else:
-            self._prefix = prefix
+            self._prefixes = set(prefix)
         self._build_commands()
 
-    @property  # type: ignore[override]
-    def command(self) -> List[str]:  # type: ignore[override]
-        """
-        The list of commands this handler should listen for.
-
-        Returns:
-            List[:obj:`str`]
-        """
-        return self._command
-
-    @command.setter
-    def command(self, command: Union[str, List[str]]) -> None:
-        if isinstance(command, str):
-            self._command = [command.lower()]
-        else:
-            self._command = command
+    @CommandHandler.commands.setter  # type: ignore[attr-defined,misc]
+    def commands(self, command: SLTS) -> None:
+        self._parse_commands(command)
```

Comment:
IISC this doesn't rebuilt `self._combinations`, but that's needed. I guess it should be `self._build_commands` instead. If you want to reduce code duplication some more, you can add a method `_slts_to_set` that returns `{arg.lower()} if isinstance(arg, str) else {item.lower() for item in arg}`. This can be a static method of `CommandHandler` or even just a module level function. You can then use it in `CH._parse_commands`, in the setter of `prefixes` and here.